controllers/utils/policy_util.go

```diff
@@ -106,3 +106,12 @@ func DeletePlacementRules(ctx context.Context, c client.Client, ns string, label
 func GetResourceName(clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade, initialString string) string {
 	return strings.ToLower(clusterGroupUpgrade.Name + "-" + initialString)
 }
+
+// GetParentPolicyNameAndNamespace gets the parent policy name and namespace from a given child policy
+// returns: []string       a two-element slice which the first element is policy namespace and the second one is policy name
+func GetParentPolicyNameAndNamespace(childPolicyName string) []string {
+	// The format of a child policy name is parent_policy_namespace.parent_policy_name.
+	// Extract the parent policy name and namespace by splitting the child policy name into two substrings separated by "."
+	// and we are safe to split with the separator "." as the namespace is disallowed to contain "."
+	return strings.SplitN(childPolicyName, ".", 2)
```

Comment:
So namespaces can't contain `.`?

Reply:
> So namespaces can't contain `.`?

Yes, namespaces can only contain lower case alphanumeric characters or '-'.  This is due to constraint of the label "kubernetes.io/metadata.name" auto-added to the namespace. Also tried with --validate=false, same result. We should be safe here.

oc create ns ztp-common-4.9 --validate=false
The Namespace "ztp-common-4.9" is invalid: metadata.name: Invalid value: "ztp-common-4.9": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')